src/eduid_webapp/group_management/views/group.py

```diff
@@ -176,5 +177,9 @@ def remove_user(user: User, group_identifier: UUID, user_identifier: UUID, role:
     except EduIDDBError:
         return error_message(CommonMsg.temp_problem)
 
-    current_app.stats.count(name=f'removed_{role.value}')
+    if user_identifier == scim_user.scim_id:
+        # If the user initiates the removal count it as "left the group"
+        current_app.stats.count(name=f'left_{role.value}')
```

Comment:
Does stat name left_member or left_owner make sense? I guess you can't forfeit ownership (?) so left_group would perhaps be better?

Also, maybe the function would be easier to read if you assigned

  _removing_self = user_identifier == scim_user.scim_id

at the top of the function? So the if-clauses would read as

  if not _removing_self and not is_owner(scim_user, group_identifier):

and

  if _removing_self: